### PR TITLE
Track retirement in explorer stake pool data

### LINKED ISSUE #2074: Unregistered pools are still listed in the explorer
Unregistered pools are still visible on the explorer, it seems the certificate is not visible

### PR DESCRIPTION
**Pull request by danielSanchezQ**
Candidate to fix #2074

### CODE REVIEW

File: jormungandr/src/explorer/graphql/mod.rs [hunk 1]
```diff
@@ -926,6 +926,30 @@ impl Pool {
...
     }
+
+    pub fn retirement(&self, context: &Context) -> FieldResult<PoolRetirement> {
+        match &self.data {
+            Some(data) => data
+                .retirement
+                .clone()
+                .map(|retirement| PoolRetirement::from(retirement))
+                .ok_or(
+                    ErrorKind::NotFound("Stake pool has no retirement data associated".to_owned())
```
Comment (enzoc4) [inline jormungandr/src/explorer/graphql/mod.rs:937]: I think it's better to return `None` here (null in graphql). And change the return type to `FieldResult<Option<PoolRetirement>>`.
Currently `Error::NotFound` is being used mostly for search-by-id queries.

Comment (danielSanchezQ) [reply jormungandr/src/explorer/graphql/mod.rs:937]: Makes sense, will do.

Comment (danielSanchezQ) [reply jormungandr/src/explorer/graphql/mod.rs]: Makes sense. I updated it. Thanks

**Comment by NicolasDP** [comment]
Can you review @enzoc4 ?

**Pull request merged**